tests/func/test_threads.py

```diff
@@ -49,3 +51,53 @@ def worker(tid, offset):
 
         session.send_request('continue').wait_for_response()
         session.wait_for_exit()
+
+
+def test_debug_this_thread(pyfile, run_as, start_method):
+    @pyfile
+    def code_to_debug():
+        from dbgimporter import import_and_enable_debugger
+        import_and_enable_debugger()
+
+        import platform
+        import ptvsd
+        import threading
+
+        def foo(x):
+            ptvsd.debug_this_thread()
+            event.set() #@bp
+            return 0
+
+        event = threading.Event()
+
+        if platform.system() == 'Windows':
+            from ctypes import CFUNCTYPE, c_void_p, c_size_t, c_uint32, windll
+            thread_func_p = CFUNCTYPE(c_uint32, c_void_p)
+            thread_func = thread_func_p(foo) # must hold a reference to wrapper during the call
+            assert windll.kernel32.CreateThread(c_void_p(0), c_size_t(0), thread_func, c_void_p(0), c_uint32(0), c_void_p(0))
+        elif platform.system() == 'Linux' or platform.system() == 'Darwin':
+            from ctypes import CDLL, CFUNCTYPE, byref, c_void_p, c_ulong
+            from ctypes.util import find_library
+            libpthread = CDLL(find_library('libpthread'))
+            thread_func_p = CFUNCTYPE(c_void_p, c_void_p)
+            thread_func = thread_func_p(foo) # must hold a reference to wrapper during the call
+            assert not libpthread.pthread_create(byref(c_ulong(0)), c_void_p(0), thread_func, c_void_p(0))
+        else:
+            pytest.skip('Not supported on ' + platform.system())
```

Comment:
pytest is not imported in this block.

Reply:
Oh, in fact this wouldn't work even if it's imported, since this isn't running in the test process. I'll move the check outside.